test/main_test.go

```diff
@@ -399,3 +310,577 @@ func (suite *JsonHttpApiSuite) TestMiddlewares_NoPayload() {
   assert.Equal(suite.T(), http.StatusUnauthorized, recorder.Code)
   assert.Equal(suite.T(), "", recorder.Body.String())
 }
+
+func (suite *JsonHttpApiSuite) TestCORS_WithoutAnyHandler() {
+  request, _ := http.NewRequest(http.MethodOptions, "/resources", nil)
+  recorder := httptest.NewRecorder()
+
+  suite.handler.ServeHTTP(recorder, request)
+
+  assert.Equal(
+    suite.T(),
+    0,
+    len(recorder.Header()[jsonContentTypeHeader]),
```

Comment:
TODO: Add tests for the other headers as well.

Reply:
Done in https://github.com/renra/go-json-http-handler/pull/10/commits/76164b046acedcf8f7ab47d7eee429d68e46405e